### PR TITLE
Added entries to explain expression-oriented languages

### PR DESCRIPTION
**Pull request by Dangthrimble**
Added definitions for 'Expression', 'Expression-Oriented Language' and 'Statement' to glossary.
Sorted the definitions alphabetically.

r? @steveklabnik

### CODE REVIEW

**Comment by Dangthrimble** [comment]
Added the fact that expression statements can form part of larger expressions.

**Comment by Dangthrimble** [comment]
Tidied up glossary.md and added link from hello-world.md to 'expression-oriented language' in glossary.md

File: src/doc/trpl/glossary.md [hunk 1]
```diff
@@ -37,3 +25,41 @@ And `2 + (3 * 4)` would look like this:
...
+
+In computer programming, an expression is a combination of values, constants,
+variables, operators and functions that evaluate to a single value. For example,
+`2 + (3 * 4)` is an expression that returns the value 14. It is worth noting
+that expressions can have side-effects. For example, a function included in an
+expression might perform actions other than simply returning a value.
+
+### Expression-Oriented Language
+
+In early programming languages [expressions][expression] and
```
Comment (tshepang) [inline src/doc/trpl/glossary.md]: missing comma between *languages* and *expression*

File: src/doc/trpl/glossary.md [hunk 2]
```diff
@@ -37,3 +25,41 @@ And `2 + (3 * 4)` would look like this:
...
+[statements][statement] were two separate syntactic categories: expressions had
+a value and statements did things. However, later languages blurred this
+distinction, allowing expressions to do things and statements to have a value.
+In an expression-oriented language, (nearly) every statement is an expression
+and therefore returns a value. Consequently these expression statements can
```
Comment (tshepang) [inline src/doc/trpl/glossary.md]: missing comma between *Consequently* and *these*

File: src/doc/trpl/glossary.md [hunk 3]
```diff
@@ -37,3 +25,41 @@ And `2 + (3 * 4)` would look like this:
...
+`2 + (3 * 4)` is an expression that returns the value 14. It is worth noting
+that expressions can have side-effects. For example, a function included in an
+expression might perform actions other than simply returning a value.
+
+### Expression-Oriented Language
+
+In early programming languages [expressions][expression] and
+[statements][statement] were two separate syntactic categories: expressions had
+a value and statements did things. However, later languages blurred this
+distinction, allowing expressions to do things and statements to have a value.
```
Comment (tshepang) [inline src/doc/trpl/glossary.md]: Do you have a citation for the claim about early vs. later programming languages? Alternatively, can you just give examples of those languages instead?

**Comment by tshepang** [comment]
Thanks for adding to the Glossary. It needs lots of love.

File: src/doc/trpl/glossary.md (hunk 3 above)
Comment (Dangthrimble) [inline src/doc/trpl/glossary.md]: Wish I could provide a citation. I didn't know what expression-oriented languages were and hoped the glossary would help. In the end I read a number of web pages (including Wikipedia!) and my description was the best I could put together. If there are inaccuracies, I just hope someone in the know corrects them!

**Comment by steveklabnik** [comment]
Thanks so much, this is great :)

**Pull request merged**